c++/src/kj/async-io-unix.c++

```diff
@@ -85,7 +85,7 @@ void setCloseOnExec(int fd) {
 }
 
 static constexpr uint NEW_FD_FLAGS =
-#if __linux__ && !__BIONIC__
+#if __linux__ && KJ_USE_FIBERS
```

Comment:
It looks like you assumed every instance of `!__BIONIC__` is guarding fiber code, but there are actually many reasons why we check for Bionic in different parts of the code. Most of these ifdefs are unrelated to fibers. This one here actually only matches Linux, so obviously it doesn't need to be updated to support OpenBSD. Can you remove the changes that aren't fiber-related?

In fact, nothing in this file is fiber-related.

Reply:
Thanks for letting me know that, I restricted the scope the KJ_USE_FIBERS and reverted the change done on this file. Please take a look :)